### ISSUE TITLE
UnmodifiableSequencedSet not supported

### ISSUE THREAD

**Issue by ThanksForAllTheFish** [kotest/kotest#3996]
**Which version of Kotest are you using**

5.8.1

Related to: https://github.com/kotest/kotest/issues/3848

Java Version 21 also introduced unmodifiable wrapper for  `SequencedSet`

When trying to test such a reverse-ordered view with shouldContainExactly it fails with an Assertion Error.

```
linkedSetOf("a", "b", "c").shouldContainExactly("a", "b", "c") // succeeds
Collections.unmodifiableSequencedSet(linkedSetOf("a", "b", "c")).shouldContainExactly("c", "b", "a") // fails
Collections.unmodifiableCollection(linkedSetOf("a", "b", "c")).shouldContainExactly("c", "b", "a") // succeeds
```

The issue probably is that the fix for 3848 still checks for subclasses of `SequencedSet` (https://github.com/kotest/kotest/pull/3873/files#diff-2d15d04b11a0d4ac020e2406ffcaa01ce78d750e0222572a224b7f54e3478309R8), I guess in an attempt to be compatible with older version of Java. 

What I saw in other project is to perform different checks for different versions

**Comment by mirageoasis** [comment]
think there is only two way
1. `gradle 8.2.1` support java 21
2. upgrade gradle

**Comment by mirageoasis** [comment]
> think there is only two way
> 
> 1. `gradle 8.2.1` support java 21
> 2. upgrade gradle

oh I found out gradle on kotest main is 8.6.1
for 8.2.1 I think it is impossible to fix but for main version I think I can support java21

**Comment by mirageoasis** [comment]
I think I can fix this issue within a week can you give me a time?

think test code will come out within a week

**Comment by Kantis** [comment]
Won't using Java 21 classes require us to use jvmTarget = 21? Basically meaning that anyone that uses Kotest needs to use Java 21?

**Comment by mirageoasis** [comment]
> Won't using Java 21 classes require us to use jvmTarget = 21? Basically meaning that anyone that uses Kotest needs to use Java 21?

oh... I was not careful enough then is there any other way around? Perhaps different code based on version

**Comment by mirageoasis** [comment]
@Kantis is there some code I can look around in this repostory? dealing with different java version?

**Comment by Kantis** [comment]
What is the expected in the case of 

```kotlin
Collections.unmodifiableSequencedSet(linkedSetOf("a", "b", "c")).shouldContainExactly("c", "b", "a") // fails
```

Given that linked sets retain order, and `shouldContainExactly` asserts for order, is it not reasonable that it fails?

**Comment by LeoColman** [comment]
Is this still an issue? Could you comment on it @mirageoasis?
I'm willing to close this issue

**Comment by mirageoasis** [comment]
> Is this still an issue? Could you comment on it @mirageoasis? I'm willing to close this issue

yes I am looking for solution but it seems I cannot find solution can some one help me with this?

**Comment by LeoColman** [comment]
Did you check @Kantis latest comment on the draft PR?

**Comment by mirageoasis** [comment]
> Did you check @Kantis latest comment on the draft PR?

oh there were some mistake in communicatino can you give me a month to solve this issue?

**Comment by LeoColman** [comment]
Sure thing! It's yours
